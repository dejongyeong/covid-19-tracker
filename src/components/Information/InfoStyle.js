import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  padding: 15px 45px;
  margin: 0 auto;

  @media only screen and (min-width: 1920.02px) {
    height: 100vh;
  }

  @media only screen and (orientation: portrait) and (min-height: 1024.98px) {
    height: 89.4vh;
  }

  @media only screen and (orientation: portrait) and ${(props) =>
      props.theme.breakpoints.md} {
    height: 100%;
    margin: 30px auto;
    padding: 15px 20px;
  }
`;

export const Header = styled.div`
  color: ${(props) => props.theme.colors.primaryColor};
  font-family: ${(props) => props.theme.font.heading};
  font-weight: 600;
  width: 100%;
  text-transform: uppercase;
  font-size: clamp(3rem, 20vw, 5rem);
  margin-top: 5px;
`;

export const Content = styled.div`
  width: 45%;
  padding: 5px 5px 5px 4px;
  font-family: ${(props) => props.theme.font.paragraph};
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 30px;
  margin: 15px 0 20px 0;
  color: ${(props) => props.theme.colors.primaryColor};

  span {
    color: ${(props) => props.theme.colors.secondaryColor};
  }

  a {
    text-decoration: none;
    color: ${(props) => props.theme.colors.tertiaryColor};

    :hover {
      color: ${(props) => props.theme.colors.primaryColor};
      text-decoration: underline;
    }
  }

  /* ts-ignore */
  @media only screen and ${(props) => props.theme.breakpoints.lg} {
    width: 100%;
  }
`;

export const InfoWrapper = styled.div`
  width: 100%;
  height: auto;
  padding: 10px 0 10px 0;
  display: flex;
  flex-direction: row;
  margin: 40px 0 20px 0;
  .card:nth-child(2) {
    margin: 0 60px;
  }

  /* ts-ignore */
  @media only screen and ${(props) => props.theme.breakpoints.lg} {
    .card:nth-child(2) {
      margin: 0 20px;
    }
  }

  @media only screen and (max-width: 807.98px) {
    flex-direction: column;
    justify-content: center;
    align-items: center;
    .card:nth-child(2) {
      margin: 20px auto;
    }
  }
`;

export const InfoCard = styled.div`
  background-color: ${(props) => props.theme.colors.primaryColor};
  font-family: ${(props) => props.theme.font.heading};
  color: ${(props) => props.theme.colors.secondaryLight};
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  max-width: 430px;
  height: 463px;
  max-height: 463px;
  border-radius: 15px;
  padding: 40px 20px 30px 20px;

  @media only screen and ${(props) => props.theme.breakpoints.xl} {
    transition: transform 0.3s ease-out;
    :hover {
      transform: translate(0, -10px);
      -webkit-transform: translate(0, -10px);
    }
  }
`;

export const InfoCardHeader = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  img {
    color: ${(props) => props.theme.colors.secondaryColor};
    width: 100%;
    max-width: 84px;
    height: auto;
  }
  h2 {
    margin-top: 12px;
  }
  h4 {
    margin-top: 5px;
    text-transform: uppercase;
    font-style: normal;
    font-weight: normal;
    font-size: 0.85rem;
    line-height: 25px;
  }
`;

export const InfoCardContent = styled.div`
  color: ${(props) => props.theme.colors.primaryLight};
  text-align: center;
  margin-top: 45px;
  margin-bottom: 20px;
  font-style: normal;
  font-weight: 500;
  font-size: 1.1rem;
  line-height: 1.5rem;
  li {
    list-style: none;
    line-height: 2rem;
  }
  span {
    font-style: normal;
    font-weight: bold;
    font-size: 48px;
    line-height: 25px;
    color: ${(props) => props.theme.colors.tertiaryColor};
  }
`;
