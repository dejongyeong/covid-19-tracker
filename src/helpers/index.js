import React from "react";
import renderer from "react-test-renderer";
import { number } from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowUp, faArrowDown } from "@fortawesome/free-solid-svg-icons";
import Theme from "../theme/Theme";

// Theme test helpers
// Tutorial: Tutorial 1: https://medium.com/styled-components/effective-testing-for-styled-components-67982eb7d42b (2017)
export function renderWithTheme(component) {
  return renderer.create(<Theme>{component}</Theme>);
}

export function calIncreasePercentage(latest, increaseValue) {
  const previous = latest - increaseValue;
  const percent = ((latest - previous) / previous) * 100;
  return percent;
}

export function getIcon(percent) {
  return percent > 0 ? (
    <FontAwesomeIcon icon={faArrowUp} />
  ) : (
    <FontAwesomeIcon icon={faArrowDown} />
  );
}

export function calPositivityRate(totalConfirm, totalTested) {
  const pr = (totalConfirm / totalTested) * 100;
  return pr;
}

export function numberWithCommas(num) {
  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
}
numberWithCommas.propTypes = {
  num: number,
};
