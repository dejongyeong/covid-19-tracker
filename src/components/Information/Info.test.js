import React from "react";
import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithTheme } from "../../helpers";

import { InfoCard, InfoWrapper } from "./InfoStyle";

afterEach(cleanup);

test("<InfoCard> test", () => {
  const tree = renderWithTheme(<InfoCard />).toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveStyleRule("transition", "transform 0.3s ease-out", {
    media: `only screen and (min-width:1200px)`,
  });
});

test("<InfoWrapper> test", () => {
  const tree = renderWithTheme(<InfoWrapper />).toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree).toHaveStyleRule("flex-direction", "column", {
    media: `only screen and (max-width: 807.98px)`,
  });

  expect(tree).toHaveStyleRule("align-items", "center", {
    media: `only screen and (max-width: 807.98px)`,
  });

  expect(tree).toHaveStyleRule("justify-content", "center", {
    media: `only screen and (max-width: 807.98px)`,
  });
});
