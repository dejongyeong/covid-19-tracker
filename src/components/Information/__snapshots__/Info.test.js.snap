// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard> test 1`] = `
.c0 {
  background-color: #003049;
  font-family: Rubik,sans-serif;
  color: #eae2b7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 430px;
  height: 463px;
  max-height: 463px;
  border-radius: 15px;
  padding: 40px 20px 30px 20px;
}

@media only screen and (min-width:1200px) {
  .c0 {
    -webkit-transition: -webkit-transform 0.3s ease-out;
    -webkit-transition: transform 0.3s ease-out;
    transition: transform 0.3s ease-out;
  }

  .c0:hover {
    -webkit-transform: translate(0,-10px);
    -ms-transform: translate(0,-10px);
    transform: translate(0,-10px);
    -webkit-transform: translate(0,-10px);
  }
}

<div
  className="c0"
/>
`;

exports[`<InfoWrapper> test 1`] = `
.c0 {
  width: 100%;
  height: auto;
  padding: 10px 0 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 40px 0 20px 0;
}

.c0 .card:nth-child(2) {
  margin: 0 60px;
}

@media only screen and (max-width:calc(992px - 0.02px)) {
  .c0 .card:nth-child(2) {
    margin: 0 20px;
  }
}

@media only screen and (max-width:807.98px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c0 .card:nth-child(2) {
    margin: 20px auto;
  }
}

<div
  className="c0"
/>
`;
